lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3269,32 +3269,52 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
 } // namespace
 
 namespace {
-class ConvertAtenOnesOp : public OpConversionPattern<AtenOnesOp> {
-public:
-  using OpConversionPattern::OpConversionPattern;
+// Converts AtenOnesOp and AtenZerosOp.
+struct ConvertAtenOnesZerosOp : ConversionPattern {
+  ConvertAtenOnesZerosOp(TypeConverter &typeConverter, MLIRContext *context)
+      : ConversionPattern(typeConverter, MatchAnyOpTypeTag(), /*benefit=*/1,
+                          context) {}
+
   LogicalResult
-  matchAndRewrite(AtenOnesOp op, OpAdaptor adaptor,
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
                   ConversionPatternRewriter &rewriter) const override {
+    if (!isa<AtenOnesOp, AtenZerosOp>(op))
+      return rewriter.notifyMatchFailure(op,
+                                         "not a supported ones or zeros op");
+
     if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
       return failure();
-    Location loc = op.getLoc();
+    Location loc = op->getLoc();
+
+    SmallVector<Value, 3> opArguments;
+    int64_t elementValue;
+
+    if (AtenOnesOp onesOp = dyn_cast<AtenOnesOp>(op)) {
+      opArguments.insert(opArguments.end(),
+                         {onesOp.size(), onesOp.layout(), onesOp.pin_memory()});
+      elementValue = 1;
+    } else if (AtenZerosOp zerosOp = dyn_cast<AtenZerosOp>(op)) {
+      opArguments.insert(opArguments.end(), {zerosOp.size(), zerosOp.layout(),
+                                             zerosOp.pin_memory()});
+      elementValue = 0;
+    }
 
     // We ignore device, but add simple asserts for unimplemented kwargs
-    if (!op.layout().getType().isa<Torch::NoneType>())
+    if (!opArguments[1].getType().isa<Torch::NoneType>())
```

Comment:
I'm not sure how I feel about using this approach. It feels like we are losing a bit of readability because now we don't know what we are getting the type of without going back to the definition of `opArguments`. You could define variables

```
Value size;
Value layout;
// etc
```

and then assign to them in the `if-else` rather than using the vector.